### PR TITLE
ci: fix whatchanged script matching

### PR DESCRIPTION
**Pull request by nashif**
When files were added, matching did not work and we were counting added
files twice. Use sets to make sure we have a unique list of files.

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -26,7 +26,7 @@ else()
   message(STATUS "West: not found")
 endif()
 
-find_package(PythonInterp 3.6 REQUIRED)
+find_package(PythonInterp 3 REQUIRED)
```
Comment (galak) [inline doc/CMakeLists.txt:29]: why the change here?  seems unrelated.

Comment (nashif) [reply doc/CMakeLists.txt]: yep, had a dirty tree...

**Review by galak — APPROVED** [review]
